src/components/PageDefinitions.js

```diff
@@ -186,31 +187,45 @@ class PageDefinitions extends Component {
     this.setState({ ...this.state, activeFacets })
   }
 
+  incrementSequence() {
+    this.setState({ ...this.state, counter: this.state.counter + 1 })
+  }
+
   noRowsRenderer() {
     return <div>Select components from the list above ...</div>
   }
 
-  renderContributeButton() {
+  collapseComponent(component) {
+    this.onChangeComponent(component, { ...component, expanded: false })
```

Comment:
`onChangeComponent` feels like it might be heavyweight here. Not sure but the folks passing in that function may (or may not) expect it to be called for expand/collapse as the component itself is not changing.